src/System.Private.CoreLib/src/System/Utf8StringSplitOptions.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    [Flags]
+    public enum Utf8StringSplitOptions
+    {
+        None = 0,
+        RemoveEmptyEntries = 1,
+        TrimEntries = 2
```

Comment:
What does `TrimEntries` do exactly and why it is worth having it for Utf8String and not regular string? It does not look right that we have special enum for Utf8String.

Reply:
It trims whitespace in addition to splitting since it's common to use `Split` and `Trim` together, especially when dealing with comma-delimited data. I'd love to include this option as part of the regular `StringSplitOptions` enum (and I've been experimenting with it separately), but that would incur significant code churn in the `String` class and would require a separate API review. My plan is to try to get that approved separately, then delete this custom enum and just use `StringSplitOptions` everywhere.

Reply:
It would be nice to add what you just said as `TODO_UTF8STRING`